### PR TITLE
HKG: Car Port for Genesis GV70 Electrified 2023 (HDA2)

### PR DESCRIPTION
**Pull request by sunnyhaibin**
**Checklist**

- [x] added entry to CAR in selfdrive/car/*/values.py and ran `selfdrive/car/docs.py` to generate new docs
- [x] test route added to [routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py)
- [x] route with openpilot: `afe09b9f5d3f3548/00000011--15fefe1c50`
- [x] route with openpilot longitudinal: `afe09b9f5d3f3548/0000001b--a1129a4a15` 
- [x] route with stock system: `afe09b9f5d3f3548/00000005--fe8cd35e57`
- [x] car harness used (if comma doesn't sell it, put N/A): `Harness Q`

Thanks to community 2023 Genesis GV70 Electrified (HDA2) owner gv70e_81046 (Discord).

**Requires**
- https://github.com/commaai/opendbc/pull/1280

Superseded https://github.com/commaai/openpilot/pull/32765

### CODE REVIEW

**Comment by jyoung8607** [comment]
Looks like the ACCELERATOR message is common to all CAN-FD EVs. Does it only have gearshift position data on this car, or might it be available on all the others? Might be an opportunity here to reduce complexity rather than increase it.

Also:

> - [x] test route added to [routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py)

This is not done, will need you to add it.

**Comment by Net1Pro** [comment]
Hello,

Please forgive my ignorance but I am not sure if the " This is not done,
will need you to add it" (uploading test route) was directed to me or
someone else.
Please advise.
Thanks!

**Comment by sunnyhaibin** [comment]
> Looks like the ACCELERATOR message is common to all CAN-FD EVs. Does it only have gearshift position data on this car, or might it be available on all the others? Might be an opportunity here to reduce complexity rather than increase it.

The signals in the `ACCELERATOR` message are only relevant for CAN-FD EV platforms, and for this car, the gearshift position data is not found in `GEAR_ALT` or `GEAR_ALT_2`. It's one of the unique quirks of HKG CAN-FD across different models and model years.

> * [x]  test route added to [routes.py](https://github.com/commaai/openpilot/blob/master/selfdrive/car/tests/routes.py)
> 
> This is not done, will need you to add it.

Added back!

**Comment by jyoung8607** [comment]
Going to mark this as draft, pending a slight refactor to cleanly support all CAN-FD EV vehicles the same way.

File: opendbc/car/hyundai/values.py [hunk 1]
```diff
@@ -527,6 +527,16 @@ class CAR(Platforms):
     CarSpecs(mass=1950, wheelbase=2.87, steerRatio=14.6),
     flags=HyundaiFlags.RADAR_SCC,
   )
+  # 2023 GV70 Electrified has Variable Gear Ratio (VGR) rack & pinion with rack mounted EPS.
+  # The overall gear/steering ratio is 13.27.
+  # https://www.genesis.com/content/dam/genesis-p2/au/assets/models/electrified-gv70/specs/Genesis_GV70_EV_August_2022_v0.1.pdf
```
Comment (jyoung8607) [inline opendbc/car/hyundai/values.py:532]:
```suggestion
```

File: opendbc/car/hyundai/values.py [hunk 2]
```diff
@@ -527,6 +527,16 @@ class CAR(Platforms):
...
+  GENESIS_GV70_ELECTRIFIED_1ST_GEN = HyundaiCanFDPlatformConfig(
+    [
+      HyundaiCarDocs("Genesis GV70 Electrified (with HDA II) 2023", "Highway Driving Assist II", car_parts=CarParts.common([CarHarness.hyundai_q])),
+    ],
+    CarSpecs(mass=2295, wheelbase=2.87, steerRatio=13.27),
```
Comment (jyoung8607) [inline opendbc/car/hyundai/values.py:537]: Adjusting steerRatio to online learned value, slight tweak to mass per spec from Edmunds.

```suggestion
    CarSpecs(mass=2260, wheelbase=2.87, steerRatio=17.1),
```

image

File: opendbc/car/torque_data/override.toml [hunk 3]
```diff
@@ -65,6 +65,7 @@ legend = ["LAT_ACCEL_FACTOR", "MAX_LAT_ACCEL_MEASURED", "FRICTION"]
 "HYUNDAI_CUSTIN_1ST_GEN" = [2.5, 2.5, 0.1]
 "LEXUS_GS_F" = [2.5, 2.5, 0.08]
 "HYUNDAI_STARIA_4TH_GEN" = [1.8, 2.0, 0.15]
+"GENESIS_GV70_ELECTRIFIED_1ST_GEN" = [2.5, 2.5, 0.1]
```
Comment (jyoung8607) [inline opendbc/car/torque_data/override.toml:68]: Adjusting lateral accel control parameters to match online learned performance.

```suggestion
"GENESIS_GV70_ELECTRIFIED_1ST_GEN" = [1.9, 1.9, 0.09]
```

image

**Review by jyoung8607 — CHANGES_REQUESTED** [review]

**Comment by jyoung8607** [comment]
GTM!

* Safety and openpilot operation equivalent to all the other HKG CAN-FD EVs
* Lateral accel control performance is reasonable and meets guidelines
* No complaints of the common HKG issue with HUD grasp-wheel/take-over alerts

Offline validation:

* Test routes were already uploaded
* `pytest opendbc` passes
* `pytest selfdrive/car/tests` passes

Caveats:

* We're not seeing the VIN, but we're also not getting it from the comma fleet EV6

**Pull request merged**